x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/settings/GetSecuritySettingsAction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.settings;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.MasterNodeReadRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+import static org.elasticsearch.xpack.core.security.action.settings.UpdateSecuritySettingsAction.MAIN_INDEX_NAME;
+import static org.elasticsearch.xpack.core.security.action.settings.UpdateSecuritySettingsAction.PROFILES_INDEX_NAME;
+import static org.elasticsearch.xpack.core.security.action.settings.UpdateSecuritySettingsAction.TOKENS_INDEX_NAME;
+
+public class GetSecuritySettingsAction extends ActionType<GetSecuritySettingsAction.Response> {
+
+    public static final GetSecuritySettingsAction INSTANCE = new GetSecuritySettingsAction();
+    public static final String NAME = "cluster:admin/xpack/security/settings/get";
```

Comment:
It is true, but `allow_restricted_indices` is a bit of a special case - it's a bit of an escape hatch to allow reaching into bits of the system we don't normally permit, rather than something we expect to be commonly used. We definitely don't want to make this rely on an index permission - that's part of the idea of system indices, the typical cluster admin shouldn't have to know about the actual underlying indices as much as possible, and definitely shouldn't have to write configuration around them - in general,we want Elasticsearch features to be managed through APIs that have stability guarantees, rather than interacting directly with system indices which do not.

You have a valid point about this API being under `manage_security` vs `manage` though. This is exactly managing security, but in a way in which it might be permissible for a "normal" cluster admin to do so. I think I could be convinced, I'll think about it while the next round of CI runs.